lib/token-assert.js

```diff
@@ -584,6 +584,14 @@ TokenAssert.prototype.noTrailingSpaces = function(options) {
                 fixed = true;
             }
 
+            if (!fixed && precendingToken && precendingToken.type === 'EOF') {
+                precendingToken = this._file.getPrevToken(precendingToken, { includeWhitespace: true });
+                if (precendingToken && precendingToken.isWhitespace) {
```

Comment:
Well, you don't need test `precendingToken` here if there is no purpose ;-).

I can't get how to break it if you remove `precendingToken`.